src/control/lib/daos/api/attribute.go

```diff
@@ -0,0 +1,268 @@
+//
+// (C) Copyright 2018-2024 Intel Corporation.
+// (C) Copyright 2025 Google LLC
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package api
+
+import (
+	"unsafe"
+
+	"github.com/pkg/errors"
+
+	"github.com/daos-stack/daos/src/control/lib/daos"
+)
+
+/*
+#include <stdint.h>
+#include <uuid/uuid.h>
+
+#include <daos_task.h>
+*/
+import "C"
+
+type attrType int
+
+const (
+	poolAttr attrType = iota
+	contAttr
+)
+
+func (at attrType) String() string {
+	switch at {
+	case poolAttr:
+		return "pool"
+	case contAttr:
+		return "container"
+	default:
+		return "unknown"
+	}
+}
+
+func listDaosAttributes(hdl C.daos_handle_t, at attrType) ([]string, error) {
+	var rc C.int
+	expectedSize, totalSize := C.size_t(0), C.size_t(0)
+
+	switch at {
+	case poolAttr:
+		rc = daos_pool_list_attr(hdl, nil, &totalSize, nil)
+	/*case contAttr:
+	rc = daos_cont_list_attr(hdl, nil, &totalSize, nil)*/
```

Comment:
Does this need to be enabled? Or removed?

Reply:
This was code moved over from the tool... The next PR (#15721) migrates a lot of the container logic into the API, and these are uncommented.